server/src/main/java/org/elasticsearch/cluster/metadata/ComposableIndexTemplate.java

```diff
@@ -294,4 +299,55 @@ public int hashCode() {
             return DataStreamTemplate.class.hashCode();
         }
     }
+    
+    public static class Builder{
+    	
+    	private List<String> indexPatterns;
+        private Template template;
+        private List<String> componentTemplates;
+        private Long priority;
+        private Long version;
+        private Map<String, Object> metadata;
+        private DataStreamTemplate dataStreamTemplate;
+        
+        
+        public Builder indexPatterns(List<String> indexPatterns) { 
+        	this.indexPatterns = indexPatterns;
+        	return this;
+        }
+        
+        public Builder template(Template template) { 
+        	this.template = template;
+        	return this;
+        }
+        
+        public Builder componentTemplates(List<String> componentTemplates) { 
+        	this.componentTemplates = componentTemplates;
+        	return this;
+        }
+        
+        public Builder priority(Long priority) { 
+        	this.priority = priority;
+        	return this;
+        }
+        
+        public Builder version(Long version) { 
+        	this.version = version;
+        	return this;
+        }
+        
+        public Builder metadata(Map<String, Object> metadata) { 
+        	this.metadata = metadata;
+        	return this;
+        }
+        
+        public Builder dataStreamTemplate(DataStreamTemplate dataStreamTemplate) { 
+        	this.dataStreamTemplate = dataStreamTemplate;
+        	return this;
+        }
+        
+        public ComposableIndexTemplate build() { 
+        	return ComposableIndexTemplate(this);
```

Comment:
The code doesn't compile, that should be `return new ComposableIndexTemplate(this)`. Also as mentioned above we should use one of the existing constructors.

Reply:
Done.